src/main/java/core/basesyntax/supplier/ShapeSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.shape.Circle;
+import core.basesyntax.shape.IsoscelesTrapezoid;
+import core.basesyntax.shape.Rectangle;
+import core.basesyntax.shape.RightTriangle;
+import core.basesyntax.shape.Shape;
+import core.basesyntax.shape.Square;
+import core.basesyntax.type.Color;
+
+public class ShapeSupplier extends BasicRandomValueSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    @Override
+    public Shape getRandom() {
```

Comment:
I have no idea how to generate a random Shape 🙈 

In Ruby language i would do somethig like this:
```
class Shape
end

class Square < Shape
end

class Rectangle < Shape
end

class Circle < Shape
end

shape = [Square, Rectangle, Circle].sample.new  # instantiate Shape subclass
shape.kind_of? Shape  # true, becaus it is subclass of Shape
```

Reply:
your approach with switch case is more than OK, don't worry :)